### ISSUE TITLE
Expression-bodied method syntax prevents mutant detection

### ISSUE THREAD

**Issue by NBlackout** [stryker-mutator/stryker-net#2798]
**Describe the bug**
No mutant is being generated for expression-bodied methods (`=> [...]` instead of `{ [...] }`)

**Logs**
[expression-body.zip](https://github.com/stryker-mutator/stryker-net/files/13672480/expression-body.zip)
[statement-body.zip](https://github.com/stryker-mutator/stryker-net/files/13672481/statement-body.zip)

**Expected behavior**
I was expecting both method syntaxes to behave similarly for mutant detection/generation.  
We use this expression-bodied syntax a lot for one-liner (props, methods, etc.) so this bug (or feature?) is quite important for us.  
The workaround would be to force statement body everywhere in our solution, or at least for project under tests.

**Desktop (please complete the following information):**
 - OS: Windows
 - Type of project: .NET
 - Framework Version: 6.0
 - Stryker Version: 3.12.0

**Additional context**
Love mutation testing, love the tool, kudos !

**Comment by NBlackout** [comment]
This issue might be a duplicate of #134 . I will check in details its content 👀

**Comment by NBlackout** [comment]
From what I understood, ternary operators are properly handled thanks to #17 (which #134 is marked duplicate of, not sure why because issues were not exactly the same even though kinda related).
I would say expression-bodied syntax handling can still be improved.

**Comment by dupdob** [comment]
Yes, can you please provide more explicit expectations and/or source code/samples. That would help us provide more accurate answer.
1. yes, there is no statement mutation for expression bodied method as those do not have any statement inside. We could either extend statement mutator to cover for his or add a new dedicated mutator (my favorite). Please note that different users have different expectations on what is a good mutator, so nothing is as obvious as one would like.
2. it also affect block mutator: an expression body is not a block statement
3. all other mutators are applied when relevant.

Hope this helps.

**Comment by NBlackout** [comment]
Thanks for the feedback ! I'm preparing a sample 👍

**Comment by NBlackout** [comment]
I can reproduce this issue even with a simple `{ return 3; }` and `=> 3`. You can run mutation test using the project under test `launchSettings.json`.  
Please let me know if setup is incomplete and I will make sure to include more details.

Have a great day 😁

[ExpressionBodiedMutationTests.zip](https://github.com/stryker-mutator/stryker-net/files/13683105/ExpressionBodiedMutationTests.zip)
